test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -295,13 +352,18 @@ describe('Test getQueryTypeComplexity function', () => {
                     name
                     ... on Droid {
                         primaryFunction
+                        friends(first: 3) {
+                            name
+                        }
                     }
                     ... on Human {
                         homeplanet
                     }
                 }
             }`;
-            expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(2); // Query 1 + hero/character 1)
+            mockDroidFriendsFunction.mockReturnValueOnce(3);
+            // Query 1 + hero/character 1 + 3 * friends/character 1)
+            expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(5);
```

Comment:
This looks good, also a fun challenge because you have to take the worst case of both branches.